ACadSharp/IO/DXF/DxfStreamWriter/DxfStreamWriterBase.cs

```diff
@@ -29,13 +31,28 @@ public void Write(int code, CSMath.IVector value, DxfClassMap map)
 			}
 		}
 
-		public void WriteHandle(int code, IHandledCadObject value, DxfClassMap map)
+		public void WriteCmColor(int code, Color color, DxfClassMap map = null)
 		{
-			if (value == null)
-			{
-				this.Write(code, (ulong)0, map);
+			byte[] arr = new byte[4];
+
+			if (color.IsTrueColor) {
+				arr[0] = (byte)color.B;
+				arr[1] = (byte)color.G;
+				arr[2] = (byte)color.R;
+				arr[3] = 0b1100_0000;
 			}
-			else
+			else {
+				arr[3] = 0b1100_0001;
+				arr[0] = (byte)color.Index;
```

Comment:
This encoding is not valid in DXF, if you check a the base sample you will see that the layer '0' has the color White which is listed in the index 7. 

This code gives a different number instead.

Reply:
I will check this. I tested it with several index and true colors and received the same values on writing as read.

Reply:
Hi @DomCR,
I thought I understand most of it but oviously I don't.

Here some examples (Created: DWG and DXF from AutoCAD, DXF saved with our `writeMultiLeader` method.

		Original					Written
	90 BL	-1033052384 = C26C DF20				-1033052384
	91 BL	-1073741824 = C000 0000				-1040187392 = C200 0000
  
	92 BL	-1033052384 = C26C DF20 -> True Color		-1033052384
	93 BL	-1056964608 = C100 0000 -> Index 0 = ByBlock	-1056964608

Written color values are essentiall ok, except the 91.
bit 0 of the byte 3 indicates, if set, that ist is an index color. "C" seems always to be there. What does bit 1 mean?

Reply:
[20240322_ml_with_style.zip](https://github.com/DomCR/ACadSharp/files/14835022/20240322_ml_with_style.zip)

Reply:
Here we have again this case: 0xC0000000 seems to be something special. It looks like the true color black. But bit 1 is not set. Remember you hat this value for CECOLOR.